### DISCUSSION TITLE
Mining on the pool by myself, not getting full reward.

### DISCUSSION THREAD

**Discussion by BrandonTaylorDev** [coinfoundry/miningcore#1689] [Q&A]
As title suggests; I'm mining on the pool, and no one else is mining there. I've confirmed in the database that my miner is the only one connected. When a block is hit, my miner is getting less than 40% of the reward, but it doing 100% of the work. The rest is sent to a random wallet address; it seems that the pool is generating a new address each time the block is hit. So far, I've hit 4 blocks and now I have 5 addresses.

I have no reward recipients configured, and I've tried to set one at 0%. Nothing changed.

**Comment by MiningCryptoLive** [comment]
We need more information. What coin are you mining?

**Reply by BrandonTaylorDev** [reply]
> We need more information. What coin are you mining?

I’m mining Bitoreum, and using the standard Core wallet for RPC. I used a brand new wallet, with a single address. Once a block was hit, a new address was automatically generated.

Am I incorrect in my thinking that a single miner in PPLNS system should receive the whole reward, minus pool fee and TX cost?

**Comment by MiningCryptoLive** [comment]
Yes you are incorrect. The single miner will not receive 100%

On Sat, Jun 3, 2023 at 11:19 AM Brandon Taylor ***@***.***>
wrote:

> We need more information. What coin are you mining?
>
> I’m mining Bitoreum, and using the standard Core wallet for RPC. I used a
> brand new wallet, with a single address. Once a block was hit, a new
> address was automatically generated.
>
> Am I incorrect in my thinking that a single miner in PPLNS system should
> receive the whole reward, minus pool fee and TX cost?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/oliverw/miningcore/discussions/1689#discussioncomment-6079205>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIAKRSFRZIROKW32VWI3G3XJNIYHANCNFSM6AAAAAAYRKYKJI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Randy Hasson
199 East Markison Avenue
Columbus, Ohio 43207

**Comment by MiningCryptoLive** [comment]
If you look through the threads on here it has been explained before. A true pplns pool will pay proportionally for the amount of shares submitted by the miner.

**Reply by BrandonTaylorDev** [reply]
> If you look through the threads on here it has been explained before. A true pplns pool will pay proportionally for the amount of shares submitted by the miner.

Even if the pool only has that one miner? That single miner performed all of the work for that block on that pool and owns 100% of the shares.

Maybe I’m misunderstanding how PPLNS works, but if that single miner owns all of the last N shares, then how is that not 1:1 proportional?

**Reply by MiningCryptoLive** [reply]
For the payments to work properly you also need to setup the wallet as a legacy wallet

**Reply by BrandonTaylorDev** [reply]
Ahhh okay, I’ll look up how that’s done and will circle back here for an update. Thanks for the suggestion.

**Reply by BrandonTaylorDev** [reply]
Circling back. I'm not seeing anything in the daemon or CLI help documentation about enabling "legacy" mode. From Bitcoin Core's legacy mode, you'd typically need to set descriptors to false, but I'm not seeing that as an option for this coin. Any advice on that?

**Reply by MiningCryptoLive** [reply]
Bitoreum - How to Setup your Wallet and Mine - BTRM - YouTube https://m.youtube.com/watch?v=Frr01iVwRyw

**Reply by BrandonTaylorDev** [reply]
Please. I can't tell if you're trolling, being an ass, or really trying to be helpful. I'm not meaning disrespect here, but at least give me some credit. I am not an idiot. Not to mention, you suggested that the wallet needs to be setup as some "legacy" type, but the video - which I watched prior to circling back a while ago - mentions nothing about.

Here is my start script, which is called by a systemd unit file. Of course, some data is scrubbed.
`/opt/crypto/wallets/bitoreum/bitoreumd -datadir="<path>" -rpcuser=user -rpcpassword=pass -daemon`

I've ran this command to generate a wallet:
`/opt/crypto/wallets/bitoreum/bitoreum-cli -datadir="<path>" -rpcuser=user -rpcpassword=pass createwallet "test_wallet"`

That's it. The output of the _createwallet_ command is simply the hash of the wallet. I configured the pool to use that hash for block rewards and have no fees configured.

The host running the MiningCore installation is a Ubuntu 22.04.

I should also point out that NOMP does not have these issues while mining similar bitcoin-family coins.

**Comment by xmrstickers** [comment]
if you hit rewards before the miner's PPLNS window is saturated, the pool keeps more than you expect in this scenario 

it's really only a "problem" for new pools - but it's functioning as expected, strictly speaking 

this thread should be helpful: 
https://bitcointalk.org/index.php?topic=39832
